inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/ProxySink.java

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    /**
+     * flush cache by batch
+     *
+     * @return thread runner
+     */
+    private Runnable coreThread() {
+        return () -> {
+            AgentThreadFactory.nameThread(
+                    "flushCache-" + profile.getTaskId() + "-" + profile.getInstanceId());
+            LOGGER.info("start flush cache {}:{}", inlongGroupId, sourceName);
+            running = true;
+            long lastPrintTime = AgentUtils.getCurrentTime();
+            while (!shutdown) {
+                try {
+                    SenderMessage senderMessage = cache.fetchSenderMessage();
+                    if (senderMessage != null) {
+                        senderManager.sendBatch(senderMessage);
+                        if (AgentUtils.getCurrentTime() - lastPrintTime > TimeUnit.SECONDS.toMillis(1)) {
+                            lastPrintTime = AgentUtils.getCurrentTime();
+                            LOGGER.info("send groupId {}, streamId {}, message size {}, taskId {}, "
+                                    + "instanceId {} sendTime is {}", inlongGroupId, inlongStreamId,
+                                    senderMessage.getDataList().size(), profile.getTaskId(),
+                                    profile.getInstanceId(),
+                                    senderMessage.getDataTime());
+                        }
+                    }
+                } catch (Exception ex) {
+                    LOGGER.error("error caught", ex);
+                } catch (Throwable t) {
+                    ThreadUtils.threadThrowableHandler(Thread.currentThread(), t);
+                } finally {
+                    AgentUtils.silenceSleepInMs(batchFlushInterval);
+                }
+            }
+            LOGGER.info("stop flush cache {}:{}", inlongGroupId, sourceName);
+            running = false;
+        };
+    }
+
+    @Override
+    public void init(InstanceProfile profile) {
+        super.init(profile);
+        fieldSplitter = profile.get(CommonConstants.FIELD_SPLITTER, DEFAULT_FIELD_SPLITTER).getBytes(
+                StandardCharsets.UTF_8);
+        sourceName = profile.getInstanceId();
+        senderManager = new SenderManager(profile, inlongGroupId, sourceName);
+        try {
+            senderManager.Start();
+            EXECUTOR_SERVICE.execute(coreThread());
+            inited = true;
+        } catch (Throwable ex) {
+            shutdown = true;
+            LOGGER.error("error while init sender for group id {}", inlongGroupId);
+            ThreadUtils.threadThrowableHandler(Thread.currentThread(), ex);
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    public void destroy() {
+        LOGGER.info("destroy sink {}", sourceName);
+        if (!inited) {
+            return;
+        }
+        shutdown = true;
+        while (running) {
+            AgentUtils.silenceSleepInMs(1);
```

Comment:
It is recommended to increase the sleep time

Reply:
ok